### PR TITLE
Auto column detection

### LINKED ISSUE #3: Automatic column detection
Hey! I thought about implementing automatic column detection to your library. It would be cool to pass a large csv without specifying each columns name. Tell me if you're interested in me implementing this functionality.

### PR DESCRIPTION
**Pull request by wojciechAdaszynski**
Closes #3 . Setting columns parameter to Boolean value  ```true``` automatically detects columns from the first row. I have also updated **a-csv** module due to some bugs.

### CODE REVIEW

**Comment by IonicaBizau** [comment]
@wojciechAdaszynski Awesome! :+1: Thanks!

File: index.js [hunk 1]
```diff
@@ -66,7 +65,10 @@ var CsvToArray = module.exports = function(options, callback) {
     options.file = String(options.file);
 
     // Validate columns (non empty array)
-    if (!options.columns || !options.columns.length || options.columns.constructor !== Array) {
+    if (options.columns === true) {
+        options.columns = [];
+    }
```
Comment (IonicaBizau) [inline index.js]: Can you put the `else` (next line) on this line?

File: index.js [hunk 2]
```diff
@@ -78,7 +80,12 @@ var CsvToArray = module.exports = function(options, callback) {
         if (err) {
             return callback(err);
         }
-
+        if (options.columns.length === 0) {
+            for (var i = 0; i < row.length; ++i) {
```
Comment (IonicaBizau) [inline index.js]: Probably you can do `options.columns = row` directly, without cloning.

**Comment by wojciechAdaszynski** [comment]
I removed array cloning but I am not sure about the else statement. Is it what you meant?

**Comment by IonicaBizau** [comment]
@wojciechAdaszynski Great. Just follow this code style:

```js
if (exp) {
   ...
} else if (exp2 || exp3) {
   ...
}
```

So, basically, put the things on the same line.
Finally don't forget to add yourself as contributor in `package.json`. :smile:

**Comment by wojciechAdaszynski** [comment]
@IonicaBizau Done :+1:

**Comment by IonicaBizau** [comment]
@wojciechAdaszynski Sorry for delay. I totally forgot about this PR. :smile:

**Pull request merged**